UnitTests/GitCommands.Tests/Git/GitRefTests.cs

```diff
@@ -0,0 +1,106 @@
+﻿using GitCommands;
+using GitUIPluginInterfaces;
+using NSubstitute;
+using NUnit.Framework;
+
+namespace GitCommandsTests.Git
+{
+    public sealed class GitRefTests
+    {
+        [Test]
+        public void IsTrackingRemote_should_return_true_when_tracking_remote()
+        {
+            string remoteBranchShortName = "remote_branch";
+            string remoteName = "origin";
+            GitRef localBranchRef = SetupLocalBranchWithATrackingReference(remoteBranchShortName, remoteName);
+
+            GitRef remoteBranchRef = SetupRemoteRef(remoteBranchShortName, remoteName);
+
+            Assert.IsTrue(localBranchRef.IsTrackingRemote(remoteBranchRef));
+        }
+
+        private static GitRef SetupRemoteRef(string remoteBranchShortName, string remoteName)
+        {
+            var remoteGitModule = Substitute.For<IGitModule>();
+            var remoteConfigFileSettings = Substitute.For<IConfigFileSettings>();
+            remoteGitModule.LocalConfigFile.Returns(remoteConfigFileSettings);
+            var remoteBranchRef = new GitRef(remoteGitModule, ObjectId.Random(), $"refs/remotes/{remoteName}/{remoteBranchShortName}", remoteName);
+            return remoteBranchRef;
+        }
+
+        private static GitRef SetupLocalBranchWithATrackingReference(string remoteShortName, string remoteName)
+        {
+            var localGitModule = Substitute.For<IGitModule>();
+            var localConfigFileSettings = Substitute.For<IConfigFileSettings>();
+            localConfigFileSettings.GetValue($"branch.local_branch.merge").Returns($"refs/heads/{remoteShortName}");
+            localConfigFileSettings.GetValue($"branch.local_branch.remote").Returns(remoteName);
+            localGitModule.LocalConfigFile.Returns(localConfigFileSettings);
+            var localBranchRef = new GitRef(localGitModule, ObjectId.Random(), "refs/heads/local_branch");
+            return localBranchRef;
+        }
+
```

Comment:
[nit] Please move these to the end of the file, after all `[Test]`s

Reply:
done.